ast/check_test.go

```diff
@@ -738,6 +743,14 @@ func TestCheckRefErrInvalid(t *testing.T) {
 			want:  types.S,
 			oneOf: []Value{String("p"), String("q")},
 		},
+		{
+			note:  "composite ref into non-set",
```

Comment:
What about type errors with terms embedded within composite operands? This is related to the comment from above.

For example...

```
p = {[1],[2],[3]}
```

And then...`p[["foo"]]` or `p[{"a": b"}]`.